backend/src/models/db/courses.js

```diff
@@ -1,17 +1,49 @@
-const data = require('../../../../frontend/public/course-names.json')
-const { toLowerCase } = require('../../utils/helpers')
-
-module.exports = data.map(item => {
-    const [courseCode, courseName] = item.split(' - ')
-
-    const facultyCode = courseCode.substr(0, 4)
-    const tags = [courseCode, courseName, facultyCode].map(toLowerCase).join(',')
-
-    return {
-        code: courseCode,
-        name: courseName,
-        facultyCode,
-        tags,
-        description: 'Offering 60 years of excellence in developing socially engaged engineers, developing new technologies and solutioning problems of global relevance.'
-    }
-})
+const data = require('../../../data/course_data_2019.json')
+const { toLowerCase, decodeUTF8Text } = require('../../utils/helpers')
+
+module.exports = (() => {
+
+        // unfortunately there are duplicate courses in the data set, so we must filter them
+
+        // list of courses to return
+        const courses = []
+
+        data.forEach(function(subj) {
+
+            const subjectCode = subj.code;
+            const subjectName = decodeUTF8Text(subj.name);
+
+            subj.courses.forEach(function(course) {
+
+                const studyLevel = course.study_level
+
+                const keywords = decodeUTF8Text(course.keywords)
+                const description = decodeUTF8Text(course.description)
+                const requirements = decodeUTF8Text(course.requirements)
+                const name = decodeUTF8Text(course.name)
+
+                const tags = [
+                    course.code,
+                    name,
+                    subjectCode,
+                    subjectName,
+                    studyLevel,
+                    ...keywords.split(',')]
+                    .map(toLowerCase).join(',')
+
+                courses.push({
+                    code: course.code,
+                    name,
+                    studyLevel,
+                    subjectCode,
+                    handbookURL: course.handbook_url,
+                    outlineURL: course.outline_url,
+                    // \n newline isn't parsed by sqlite, so replace it with html
+                    description: description.replace("\n", "<p></p>"),
```

Comment:
Should also parse list tags since it is breaking descriptions?
E.g.
`introduction to ethicsworkable ethical theoriesmorality of breaking the lawprivacypublic information and recordsand related topics Students carry out a significant software development project in teams.`